workflow/controller/operator.go

```diff
@@ -589,13 +589,18 @@ func (woc *wfOperationCtx) persistUpdates(ctx context.Context) {
 	// errors if we label a pod that was deleted already.
 	for podName, podPhase := range woc.completedPods {
 		if woc.execWf.Spec.PodGC != nil {
+			var doPodGC bool
+
 			switch woc.execWf.Spec.PodGC.Strategy {
 			case wfv1.PodGCOnPodSuccess:
-				if podPhase == apiv1.PodSucceeded {
-					woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
-				}
+				doPodGC = podPhase == apiv1.PodSucceeded
 			case wfv1.PodGCOnPodCompletion:
-				woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
+				doPodGC = true
+			}
+
+			if doPodGC {
+				delayDuration := time.Duration(woc.execWf.Spec.PodGC.DelaySeconds) * time.Second
```

Comment:
can you add a `func (p PodGC) GetDelay() time.Duration` to hide this in?

Reply:
Sure can makes sense.